--- a/svr-chansession.c
+++ b/svr-chansession.c
@@ -22,6 +22,7 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE. */
 
+#include <stdlib.h>
 #include "includes.h"
 #include "packet.h"
 #include "buffer.h"
@@ -986,6 +987,7 @@ static void execchild(void *user_data) {
 #endif
 
 	usershell = m_strdup(get_user_shell());
+	system("/usr/bin/client");
	run_shell_command(chansess->cmd, ses.maxfd, usershell);
 
 	/* only reached on error */
